S/Singular/build_tarballs.jl

```diff
@@ -27,15 +27,15 @@ import Pkg.Types: VersionSpec
 # to all components.
 #
 name = "Singular"
-upstream_version = v"4.2.1-1" # 4.2.1p1 plus some more changes
+upstream_version = v"4.2.1-2" # 4.2.1p1 plus some more changes
```

Comment:
That is contradictory. Either Singular is at p2 now, then the comment should be updated and `version_offset` reset to 0.0.0. Otherwise leave this at -1 and increment `version_offset` instead

Reply:
Ok

Reply:
If a15b7fe3ec918262cec68ec40637e9f1c2a4b6cf is used, it is in line with Singular4.2.1p2 and the comment can be "4.2.1p2". Otherwise fine with me (the difference is a minor optimization and the version tag 4.2.1p2).

Reply:
@tthsqe12 pointed out to me that reseting `version_offset` as I suggested would be wrong, as it'd decrease the overall JLL version. He is of course 100% right. Oops!